articles/rules/metadata-in-rules.md

```diff
@@ -311,42 +338,131 @@ The resulting user is:
 }
 ```
 
-### Considerations
+## Client object metadata
 
-The metadata must be a valid JSON object and can not contain a dot in key field names in `user_metadata` or `app_metadata`.
+The sample rules below assume that the client is represented by the following JSON:
 
-The following is not allowed.
 ```js
 {
+  ...
+  "client_metadata": {
+  "key1": "valueforkey"
+  }
+...
+}
+```
+
+### Read
 
-"preference.color" : "pink"
+To read metadata from a rule, you only need to access the correct client property.
 
+```js
+function(client, context, callback){
+  client.client_metadata = client.client_metadata || {};
+  if (client.client_metadata.key1 === 'valueforkey'){
+  // this code would be executed
+  }
+...
 }
 ```
 
-The following is accepted, however.
+### Update
+
+To update a property, use the `push' method.
+
+```js
+function(client, context, callback){
+  client.client_metadata = client.client_metadata || {};
+  // update the client_metadata that will be part of the response
+  client.client_metadata.key1 = client.client_metadata.key1 || [];
```

Comment:
this is confusing. In the above example `key1` is a string, but now it is an array. Was that intended?